src/main/scala/com/lightbend/lagom/core/persistence/effects/core/persistence/AccountEntity.scala

```diff
@@ -2,107 +2,118 @@ package com.lightbend.lagom.core.persistence.effects.core.persistence
+          readOnlyCommandHandlers and
+            withdrawCommandHandlers(account) and
+            depositCommandHandlers and
+            afterCreationEventsHandlers(account)
       }
 
-  // on creation and later
-  val depositCommandActions =
-    actions
-      .onCommand {
-        // this is directive builder that expects one single Event
-        Handler[Deposit]
-          .persistOne {
-            case cmd => DepositExecuted(cmd.amount)
-          }
-          .andThen {  (evt, state) =>
-            println(s"Deposit ${evt.amount}, current balance is ${state.amount}")
-          }
-          // reply with new balance
-          .replyWith(_.amount)
-      }
 
-  def withdrawCommandActions(account: Account) =
-    actions
-      .onCommand {
-        Handler[Withdraw]
-          .attempt.persistOne { // <- this Effect builder expects a Try[Event]
-            case cmd if account.amount - cmd.amount >= 0 => Success(WithdrawExecuted(cmd.amount))
-            case cmd => Failure(new RuntimeException("Insufficient balance"))
-          }
-          // NOTE: we don't need reply because Withdraw replies with Done
-          // and there is an implicit for it
-      }
+  // on creation and later
+  val depositCommandHandlers: CommandHandlers =
+    onCommand {
+      // this is directive builder that expects one single Event
+      Handler[Deposit]
+        .persistOne {
+          case cmd => DepositExecuted(cmd.amount)
+        }
+        .andThen { (evt, state) =>
+          println(s"Deposit ${evt.amount}, current balance is ${state.amount}")
+        }
+        // reply with new balance
+        .replyWith(_.amount)
+    }
+
+  def withdrawCommandHandlers(account: Account): CommandHandlers =
+    onCommand {
+      Handler[Withdraw]
+        .attempt
+        .persistOne { // <- this Effect builder expects a Try[Event]
+          case cmd =>
+            account
+              .validateWithdraw(cmd.amount) // <- this method returns a Try[Double]
+              .map(WithdrawExecuted)
+        }
+      // NOTE: we don't need reply because Withdraw replies with Done
+      // and there is an implicit for it
+    }
 
 
-  // read-only commands are no different, it's just an effect without a command handler
-  // behind the scenes, this is a regular Effect with a NoOps command handler.
-  // It doesn't have callbacks neither, but do have a reply.
-  val readOnlyCommands =
-    actions
-      .onCommand {
-        ReadOnly[GetBalance.type].replyWith(_.amount)
-      }
-      .onCommand {
-        // there is also an implicit for reply with State
-        // but intellij gives an error hence the explicit call here
-        ReadOnly[GetState.type].replyWith(identity)
-      }
+  /**
+    * read-only commands are no different, it's just an effect without a command handler
+    * behind the scenes, this is a regular Effect with a NoOps command handler.
+    * It doesn't have callbacks neither, but do have a reply.
+    */
```

Comment:
How would you model an authorization operation?
Imagine a read-only operation where I try to access your data. The command should still be validated with zero events emitted.

Reply:
Good one! I didn't though about that. 

What about...
```scala
ReadOnly[MyCommand]
   .authorise { 
     case cmd => true // state is in scope, no need to have it passed
   }
   .replyWith(something)  
```

Reply:
I don't think `authorise` is a good idea. I would implement it as a regular validation.

Reply:
You mean call it `validate` instead?

The problem with a general validation is that it usually means that if valid events are emitted, it not command is rejected. 

We have it already with `.attempt.persist()` or event with `persist()` + exception throw. 

In the case of a read-only command, we want it without emitting events.

Reply:
Ok, I got what you mean. It's about having a generic name like `validation` instead of `authorise`.  Yes, for sure.

Reply:
no, I think the validation code is part of the command handler and I would not add any validation step before that:

```scala
.persistOne { // <- this Effect builder expects a Try[Event]
            case cmd if account.amount - cmd.amount >= 0 => Success(WithdrawExecuted(cmd.amount))
            case cmd => Failure(new RuntimeException("Insufficient balance"))
          }
```

I think this code is what I prefer: I have control over the returned type in case of validation failure, I have access to the state and the command to implement my validation, I can just ignore the validation. I don't think and `authorise` or a `validation` steps should be included because I get the feeling the DSL will be very cluttered.